addons/sale_stock/static/src/xml/sale_stock.xml

```diff
@@ -1,31 +1,34 @@
 <templates>
     <div t-name="sale_stock.qtyAtDate">
         <div t-att-class="!widget.data.display_qty_widget ? 'invisible' : ''">
-            <t t-if="widget.data.virtual_available_at_date &lt; widget.data.qty_to_deliver and !widget.data.is_mto">
-                <a tabindex="0" class="fa fa-info-circle text-danger"/>
+            <t t-if="['draft', 'sent'].includes(widget.data.state) and widget.data.virtual_available_at_date &lt; widget.data.qty_to_deliver and !widget.data.is_mto">
+                <a tabindex="0" class="fa fa-area-chart text-danger"/>
+            </t>
+            <t t-elif="widget.data.state == 'sale' and ((widget.data.forecast_expected_date and widget.data.scheduled_date &lt; widget.data.forecast_expected_date) or (!widget.data.forecast_expected_date and widget.data.virtual_available_at_date &lt; widget.data.qty_to_deliver))">
```

Comment:
I think these conditions could be more readable if done in the JS.
Also, if you move conditions in the JS, you'll can do something like that:
```xml
<div t-att-class="!widget.data.display_qty_widget ? 'invisible' : ''">
    <a tabindex="0" t-class="'fa fa-area-chart text-' + widget.is_late ? 'danger' : 'primary'"/>
</div>```